siliconcompiler/tools/yosys/syn_fpga.py

```diff
@@ -38,10 +38,75 @@ def setup_fpga(chip):
              ",".join(['fpga', part_name, 'lutsize']),
              step=step, index=index)
 
+    chip.add('tool', tool, 'task', task, 'require',
+             ",".join(['fpga', part_name, 'var', 'flop_async_set']),
+             step=step, index=index)
+
+    chip.add('tool', tool, 'task', task, 'require',
+             ",".join(['fpga', part_name, 'var', 'flop_async_reset']),
+             step=step, index=index)
+
+    chip.add('tool', tool, 'task', task, 'require',
+             ",".join(['fpga', part_name, 'var', 'flop_enable']),
+             step=step, index=index)
+
+    chip.add('tool', tool, 'task', task, 'require',
+             ",".join(['fpga', part_name, 'var', 'legalize_flops']),
+             step=step, index=index)
+
+    # Do not require tech libraries, as there are some FPGAs that
+    # are so simple (in academia) that they do not require any
+
     chip.add('tool', tool, 'task', task, 'output', design + '_netlist.json', step=step, index=index)
     chip.add('tool', tool, 'task', task, 'output', design + '.blif', step=step, index=index)
 
 
+##################################################
```

Comment:
I don't think this preprocess adds any value to the readability of the tcl since it's just copying the values over.